pkgs/os-specific/darwin/install-binary-package/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, stdenvNoCC, makeSetupHook, _7zz, libarchive }:
+
+{ pname
+, version
+, src
+, buildInputs ? []
+, nativeBuildInputs ? []
+, propagatedBuildInputs ? []
+, postInstall ? null
+, appName ? "."
+, sourceRoot ? "."
+, ...
+}@args:
+
+let
+  unpackDmgPkg = makeSetupHook {
+    name = "unpack-dmg-pkg";
+    propagatedBuildInputs = [ _7zz libarchive ];
+  } ./unpack-dmg-pkg.sh;
+in
+stdenvNoCC.mkDerivation (args // {
+  inherit pname appName src buildInputs propagatedBuildInputs postInstall;
+
+  dontMakeSourcesWritable = args.dontMakeSourcesWritable or true;
+  dontPatch = args.dontPatch or true;
+  dontConfigure = args.dontConfigure or true;
+  dontBuild = args.dontBuild or true;
+  dontStrip = args.dontStrip or true;
+  dontFixup = args.dontFixup or true;
+  dontCheck = args.dontCheck or true;
+  doInstallCheck = args.doInstallCheck or false;
```

Comment:
- [ ] Is this the best way to set sensible defaults, but allow customisation for packages that need it?
- [ ] Are there any other `dont*` flags that are relevant here?